pkg/apis/ceph.rook.io/v1/annotations.go

```diff
@@ -110,14 +110,20 @@ func (a Annotations) ApplyToObjectMeta(t *metav1.ObjectMeta) {
 // original Annotations with the attributes of the supplied one. The supplied
 // Annotation attributes will override the original ones if defined.
 func (a Annotations) Merge(with map[string]string) Annotations {
-	ret := a
-	if ret == nil {
-		ret = map[string]string{}
+	// Create a new map of type Annotations to hold the merged results
+	ret := Annotations{}
+
+	// Copy the contents of the original map (a) into ret
+	for k, v := range a {
```

Comment:
Still trying to understand how the `mgr` annotations were applied to `osd`, can you explain at what part it was getting miss applied, because I see `with map[string]string` just contains the values of a particular daemon